### PR TITLE
Fix some trivial deprecation warnings #33

### PR DESCRIPTION
**Pull request by emaldona**
The scope is this patch is very narrow.

### CODE REVIEW

**Comment by emaldona** [comment]
Notice a section that  has 
+    @Deprecated
which is not a fix but adding a deprecation warning. 

This diff with 8 lines of context should make it clear why it's needed.
diff --git a/org/mozilla/jss/ssl/SSLVersionRange.java b/org/mozilla/jss/ssl/SSLVersionRange.java
index 188887c5..8823f451 100644
--- a/org/mozilla/jss/ssl/SSLVersionRange.java
+++ b/org/mozilla/jss/ssl/SSLVersionRange.java
@@ -51,16 +51,17 @@ public class SSLVersionRange {
 
     /**
      * Used by the C code, do not use it directly
      * @deprecated Replaced with SSLVersionRange(SSLVersion minVersion, SSLVersion maxVersion).
      * @param min_enum
      * @param max_enum
      * @throws IllegalArgumentException
      */
+    @Deprecated
     public SSLVersionRange(int min_enum, int max_enum) throws IllegalArgumentException {
         this(SSLVersion.valueOf(min_enum), SSLVersion.valueOf(max_enum));
     }
 
     public SSLVersion getMinVersion() {
         return minVersion;
     }

File: org/mozilla/jss/tests/JSS_SelfServServer.java [hunk 1]
```diff
@@ -112,13 +113,14 @@ public void doIt(String[] args) throws Exception {
...
             if (args.length >= 5)
-                port = new Integer(args[4]).intValue();
+                port = new Integer(Integer.valueOf(args[4]));
             if (args.length >=6 && args[5].equalsIgnoreCase("verbose")) {
                 bVerbose = true;
             }
             if (args.length >=7 && !args[6].equalsIgnoreCase("default")) {
                 fServerCertNick = args[6];
             }
+        } catch (NumberFormatException e) {
```
Comment (cipherboy) [inline org/mozilla/jss/tests/JSS_SelfServServer.java]: Out of curiousity, why adding this line? Seems like it should be caught with the next line. :)

File: org/mozilla/jss/ssl/SSLTest.java [hunk 2]
```diff
@@ -56,7 +56,7 @@ public SSLTest(String[] args) {
         dumpParams();
         CryptoManager.initialize(".");
 
-        int port = (new Integer( params.get("port") )).intValue();
+        int port = (Integer.valueOf(params.get("port")));
```
Comment (cipherboy) [inline org/mozilla/jss/ssl/SSLTest.java]: You could probably remove the outer parenthesis while you're here.

File: org/mozilla/jss/tests/JSSE_SSLClient.java [hunk 3]
```diff
@@ -274,13 +275,13 @@ private void initSocketFactory() {
                 System.setProperty("java.protocol.handler.pkgs",
                         "com.ibm.net.ssl.www.protocol.Handler");
                 java.security.Security.addProvider((java.security.Provider)
-                Class.forName("com.ibm.jsse2.IBMJSSEProvider2").newInstance());
+                Class.forName("com.ibm.jsse2.IBMJSSEProvider2").getConstructor().newInstance());
```
Comment (cipherboy) [inline org/mozilla/jss/tests/JSSE_SSLClient.java]: Looking at this, I'm not sure why we keep it around. I'm guessing that this test case probably doesn't work on the OpenJDK any more. It definitely isn't being run by our test harness but that's alright.

I'd probably update it to match the newer style (`main(...) throws Exception` and not catch the error) if I were picky, but like I said, not being used and likely won't be used, so I wouldn't bother.

Comment (emaldona) [reply org/mozilla/jss/tests/JSSE_SSLClient.java]: Agree, not worth the effort.

**Comment by emaldona** [comment]
Will submit a revised patch.

**Comment by emaldona** [comment]
Need some help, I am out of practice and forgot how to do revision to patches. I'll ask you online.

**Pull request merged**